### PR TITLE
add fluentd implementation

### PR DESCRIPTION
**Pull request by normanjoyner**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    },
+
+    initialize(core) {
+        if(!core || !core.logger) {
+            return module.exports.runCLI();
+        } else if(core.options.mode === 'leader') {
+            return module.exports.runLeader(core);
+        } else if(core.options.mode === 'follower') {
+            return module.exports.runFollower(core);
+        } else if(core.logger) {
```
Comment (nicktate) [inline index.js]: how could this ever happen?

Comment (nicktate) [inline index.js]: more like, have you already seen this happen?

Comment (normanjoyner) [inline index.js]: no, but I would rather not assume `core.options.mode` is something it is not

File: lib/implementations/fluentd.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        (err) => {
+                            if(err) {
+                                return core.applications.add({
+                                    container_port: FLUENTD_PORT,
+                                    cpus: 0.1,
+                                    env_vars: {},
+                                    id: APPLICATION_NAME,
+                                    image: 'containership/fluentd-logs:latest',
+                                    memory: 64,
+                                    network_mode: 'host',
```
Comment (nicktate) [inline lib/implementations/fluentd.js]: do we need to default to host mode?

Comment (normanjoyner) [inline lib/implementations/fluentd.js]: no, but the agent is only intended to be accessed locally and this way avoids querying myriad on each container launch

**Pull request merged**